### PR TITLE
Add PATCH method to modify resources -> moved to #2263

### PR DESCRIPTION
**Pull request by adrianlzt**
Allow a new call to the REST API, using method PATCH, to modify
resources.
The body of the request should use JSON Patch
(https://tools.ietf.org/html/rfc6902) to specify the modifications.

Only metatada could be modified.
For node resources, some fields inside metadata could not be changed.

TODO:
- [x] test noderule/edgerule modifications
- [x] test with ES backend
- [X] tests
- [ ] add to swagger

### CODE REVIEW

**Comment by lebauce** [comment]
run tests

**Comment by adrianlzt** [comment]
Example of use:
```
➜ curl 127.0.0.1:8082/api/node/test -XPATCH -d '
[
                {
                  "op": "add",
                  "path": "/Metadata/NewProp",
                  "value": "foobar"
                },
                {
                  "op": "modify",
                  "path": "/Metadata/Prop",
                  "value": "changed prop"
                },
                {
                  "op": "remove",
                  "path": "/Metadata/OldProp"
                },
]'
```

File: graffiti/graph/graph.go [hunk 1]
```diff
@@ -724,6 +725,8 @@ func (g *Graph) SetMetadata(i interface{}, m Metadata) error {
 	case *Edge:
 		e = &i.graphElement
 		kind = EdgeUpdated
+	default:
```
Comment (lebauce) [inline graffiti/graph/graph.go:728]: nit: probably not required

File: graffiti/graph/memory.go [hunk 2]
```diff
@@ -41,14 +41,19 @@ type MemoryBackend struct {
 func (m *MemoryBackend) MetadataUpdated(i interface{}) error {
 	switch i := i.(type) {
 	case *Node:
-		if _, ok := m.nodes[i.ID]; !ok {
-
+		node, ok := m.nodes[i.ID]
+		if !ok {
 			return ErrNodeNotFound
 		}
+		node.graphElement.Metadata = i.graphElement.Metadata
```
Comment (lebauce) [inline graffiti/graph/memory.go:48]: should be removed

File: graffiti/api/server/server.go [hunk 3]
```diff
@@ -206,6 +207,94 @@ func (a *Server) RegisterAPIHandler(handler rest.Handler, authBackend shttp.Auth
...
+
+				// Patch are applied to the json body in []byte format
+				resourcePatchedJSON, err := patch.Apply(resourceJSON)
+				if err != nil {
+					http.Error(w, err.Error(), http.StatusBadRequest)
+					return
+				}
+
+				resourcePatched := handler.New()
+				err = json.Unmarshal(resourcePatchedJSON, resourcePatched)
```
Comment (lebauce) [inline graffiti/api/server/server.go:261]: nit: if err := json.Unmarshal(resourcePatchedJSON, resourcePatched); err != nil {

**Comment by adrianlzt** [comment]
Moved to PR #2263 to allow maintainers modifications

**Pull request closed without merging**